test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,404 @@
+	let polkadot_bob_id = System::network_state(&mut client_bob).await.unwrap()["peerId"]
+		.as_str()
+		.unwrap()
+		.to_string();
+
+	// retrieve runtime version
+	let runtime_version = State::runtime_version(&mut client_alice).await.unwrap();
+
+	// get the current block
+	let current_block_hash = Chain::block_hash(&mut client_alice, None)
+		.await
+		.unwrap()
+		.unwrap();
+	let current_block = Chain::header(&mut client_alice, current_block_hash)
+		.await
+		.unwrap()
+		.unwrap()
+		.number
+		.saturated_into::<u64>();
+
+	let genesis_block = Chain::block_hash(&mut client_alice, 0)
+		.await
+		.unwrap()
+		.unwrap();
+
+	// create and sign transaction
+	let wasm =
+		fs::read(target_dir().join(
+			"wbuild/cumulus-test-parachain-runtime/cumulus_test_parachain_runtime.compact.wasm",
+		))
+		.unwrap();
+	let call = pallet_sudo::Call::sudo(Box::new(
+		registrar::Call::<Runtime>::register_para(
+			100.into(),
+			Info {
+				scheduling: Scheduling::Always,
+			},
+			wasm.into(),
+			genesis_state.into(),
+		)
+		.into(),
+	));
+	let nonce = 0;
+	let period = BlockHashCount::get()
+		.checked_next_power_of_two()
+		.map(|c| c / 2)
+		.unwrap_or(2) as u64;
+	let tip = 0;
+	let extra: SignedExtra = (
+		OnlyStakingAndClaims,
+		frame_system::CheckVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckEra::<Runtime>::from(generic::Era::mortal(period, current_block)),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+		registrar::LimitParathreadCommits::<Runtime>::new(),
+		parachains::ValidateDoubleVoteReports::<Runtime>::new(),
+	);
+	let raw_payload = SignedPayload::from_raw(
+		call.clone().into(),
+		extra.clone(),
+		(
+			(),
+			runtime_version.spec_version,
+			genesis_block,
+			current_block_hash,
+			(),
+			(),
+			(),
+			(),
+			(),
+		),
+	);
+	let signature = raw_payload.using_encoded(|e| Alice.sign(e));
+
+	// register parachain
+	let ex = polkadot_runtime::UncheckedExtrinsic::new_signed(
+		call.into(),
+		Alice.into(),
+		sp_runtime::MultiSignature::Sr25519(signature),
+		extra,
+	);
+	let _register_block_hash =
+		Author::submit_extrinsic(&mut client_alice, format!("0x{}", hex::encode(ex.encode())))
+			.await
+			.unwrap();
+
+	// run cumulus
+	let cumulus_dir = tempdir().unwrap();
+	let mut cumulus = Command::new(cargo_bin("cumulus-test-parachain-collator"))
+		.stdout(Stdio::piped())
+		.stderr(Stdio::piped())
+		.arg("--base-path")
+		.arg(cumulus_dir.path())
+		.arg("--")
+		.arg(format!(
+			"--bootnodes=/ip4/127.0.0.1/tcp/30333/p2p/{}",
+			polkadot_alice_id
+		))
+		.arg(format!(
+			"--bootnodes=/ip4/127.0.0.1/tcp/50666/p2p/{}",
+			polkadot_bob_id
+		))
+		.spawn()
+		.unwrap();
+	let mut cumulus_helper = ChildHelper::new("cumulus", &mut cumulus);
+
+	// wait for blocks to be generated
+	thread::sleep(Duration::from_secs(60));
```

Comment:
Never sleep in tests! ;) 

Why can we not check that the best blocks progresses using the rpc call?

Reply:
done